File: GameFramePro/Assets/.lpproj/Apis/Unity/CS/GameFramePro/ResourcesEx/UpdateAssetBundleInfor.lua
---@class CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor : CS.System.Object
CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor = {}

---@field public CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor.mAssetBundleAssetUpdateTagEnum : CS.GameFramePro.ResourcesEx.AssetBundleAssetUpdateTagEnum
CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor.mAssetBundleAssetUpdateTagEnum = nil

---@field public CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor.mNeedUpdateAssetName : CS.System.String
CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor.mNeedUpdateAssetName = nil

---@field public CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor.mAssetByteSize : CS.System.Int64
CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor.mAssetByteSize = nil

---@field public CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor.mAssetCRC : CS.System.UInt32
CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor.mAssetCRC = nil

---@return CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor
function CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor()
end

---@param tagEnum : CS.GameFramePro.ResourcesEx.AssetBundleAssetUpdateTagEnum
---@param assetName : CS.System.String
---@param byteSize : CS.System.Int64
---@param crc : CS.System.UInt32
---@return CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor
function CS.GameFramePro.ResourcesEx.UpdateAssetBundleInfor(tagEnum, assetName, byteSize, crc)
end

File: GameFramePro/Assets/Scripts/Core/Module/Upgrade/UpgradeAssetBundle/UpdateAssetBundleInfor.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using System;

namespace GameFramePro.Upgrade
{
    /// <summary>/// 标示更新Asset Bundle 资源时候的状态/// </summary>
    internal enum AssetBundleAssetUpdateTagEnum
    {
        AddAsset, //新增资源
        RemoveLocalAsset, //本地资源已经无效 需要删除
        UpdateAsset, //本地资源需要更新
    }


    [System.Serializable]
    internal class UpdateAssetBundleInfor
    {
        public AssetBundleAssetUpdateTagEnum mAssetBundleAssetUpdateTagEnum;
        public string mNeedUpdateAssetName = string.Empty;
        public long mAssetByteSize = 0;
        public uint mAssetCRC = 0;


        public UpdateAssetBundleInfor()
        {
        }

        public UpdateAssetBundleInfor(AssetBundleAssetUpdateTagEnum tagEnum, string assetName, long byteSize, uint crc)
        {
            mAssetBundleAssetUpdateTagEnum = tagEnum;
            mNeedUpdateAssetName = assetName;
            mAssetByteSize = byteSize;
            mAssetCRC = crc;
        }

        public UpdateAssetBundleInfor(AssetBundleAssetUpdateTagEnum tagEnum, AssetBundleInfor bundleInfor)
        {
            mAssetBundleAssetUpdateTagEnum = tagEnum;
            mNeedUpdateAssetName = bundleInfor.mBundleName;
            mAssetByteSize = bundleInfor.mBundleSize;
            mAssetCRC = bundleInfor.mBundleCRC;
        }
    }
}
